File: back/src/main/java/com/back/domain/trend/snapshot/repository/DailyKeywordCountRepository.java
package com.back.domain.trend.snapshot.repository;

import com.back.domain.trend.snapshot.entity.DailyKeywordCount;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDate;
import java.util.Optional;

public interface DailyKeywordCountRepository extends JpaRepository<DailyKeywordCount, Long> {

    Optional<DailyKeywordCount> findByDateAndKeyword(LocalDate date, String keyword);
}

File: back/src/main/kotlin/com/back/domain/trend/snapshot/repository/DailyKeywordCountRepository.kt
package com.back.domain.trend.snapshot.repository

import com.back.domain.trend.snapshot.entity.DailyKeywordCount
import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDate

interface DailyKeywordCountRepository : JpaRepository<DailyKeywordCount, Long> {
    fun findByDateAndKeyword(date: LocalDate, keyword: String): DailyKeywordCount?

    fun findAllByDate(date: LocalDate): List<DailyKeywordCount>
}
